students/hirot_directory/session10/oo_mailroom_hiro/mailroom.py

```diff
@@ -0,0 +1,212 @@
+#!/usr/bin/env python3
+
+import sys
+
+
+def request_donation():
+    donor_name = str(input("Full Name: "))
+    new_donation = int(input("How much do you want to donate?: "))
+    if (new_donation > 0):
+        print("Thank you for donation, Mr./Ms. {}! Your donation amount is USD ${}.".format(donor_name, new_donation))
+    return Donor(donor_name, new_donation)
+
+
+def select_submenu():
+    choice = int(input("Select from one of these options:\n"
+                       "(1) Show a list\n"
+                       "(2) Enter a donation\n"
+                       "(3) Exit\n"))
+    return choice
+
+
+def select_mainmenu():
+    answer = int(input("Select from one of these options:\n"
+                       "(1) Send a Thank you\n"
+                       "(2) Create a Report\n"
+                       "(3) Send letters to everyone\n"
+                       "(4) quit\n"))
+    return answer
+
+
+def select_quit():
+    selection = (input("Are you sure to quit? (Y or N) To return to main menu, please select 'N'.: ")).upper()
+    return selection
+
+
+def mainloop():
+    while True:
+        try:
+            answer = select_mainmenu()
+            dict_answer[answer]()
+
+        except ValueError as e:
+            print("Error happened. Error is ", type(e), "Please type 1, 2 or 3")
+        except KeyError as k:
+            print("Error happened. Error is ", type(k), "Please type 1, 2 or 3")
+
+
+def thank_you_loop():
+    while True:
+        try:
+            choice = select_submenu()
+            dict_choice[choice]()
+
+        except ValueError as e:
+            print("Error happened. Error is ", type(e) , "Please type 1, 2 or 3")
+        except KeyError as k:
+            print("Error happened. Error is ", type(k) , "Please type 1, 2 or 3")
+
+
+def quit():
+    print("quit function is called")
+
+    try:
+        selection = select_quit()
+        if (selection == 'Y'):
+            print("Good bye...")
+            sys.exit()
+        else:
+            return mainloop()
+
+    except ValueError as e:
+        print("Error happened. Error is ", type(e), "Please type 1, 2 or 3")
+    except KeyError as k:
+        print("Error happened. Error is ", type(k), "Please type 1, 2 or 3")
+
+
+def make_sample_db():
```

Comment:
now that you have a DonorDB class, this function should return one of those!

it could start by making an empty DonorDB, then loop through the names and donations, call add_donor on the DonorDB, then add the donations to the donor.